### ISSUE TITLE
Start-Process with a macOS application bundle doesn't support arguments

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#10598]
The problem seems related to the fact that macOS `.app` bundles are _directories_, not single files.

# Steps to reproduce

```powershell
# On macOS
# This should be the equivalent of:
#      open -a /Applications/Safari.app http://example.org  # or: open -a Safari http://example.org
{ Start-Process /Applications/Safari.app -ArgumentList http://example.org } |
  Should -Not -Throw
```

# Expected behavior

The test should succeed, and Safari should open to http://example.org

# Actual behavior

The test fails:

```none
Expected no exception to be thrown, but an exception "The FileName property should not be a directory unless UseShellExecute is set." was thrown from
```

# Environment data

```none
PowerShell Core v7.0.0-preview.3 on macOS 10.14.6
```

**Comment by joshtbernstein** [comment]
Still an issue in 2021. 👎

**Comment by mikebuckley** [comment]
March 2022 - Still nada.   What is up?

**Comment by mklement0** [comment]
I agree that this should be fixed, but just to mention the workaround: `open -a Safari http://example.org`

**Comment by computersarecool** [comment]
Hey @microsoft @satyanadella @PowerShellTeam despite a great bug report from @mklement0 this issue has not been addressed (nor even commented on) by anyone from your team more than three years later. This is exactly the thing that demonstrates that you are not serious about making Powershell actually cross-platform. Please fix the issue or at least comment. Thanks!

**Comment by StewartRyanSony** [comment]
Still a problem. It seems there are no plans of allowing users outside of Windows to launch applications from PowerShell
